pyqtgraph/functions.py

```diff
@@ -1492,18 +1492,29 @@ def arrayToQPath(x, y, connect='all'):
     arr[1:-1]['x'] = x
     arr[1:-1]['y'] = y
 
+    non_finite_values = np.logical_not(np.isfinite(x) & np.isfinite(y))
+
+    arr[1:-1]['x'][non_finite_values] = np.interp(np.flatnonzero(non_finite_values), np.flatnonzero(~non_finite_values), arr[1:-1]['x'][~non_finite_values])
+    arr[1:-1]['y'][non_finite_values] = np.interp(np.flatnonzero(non_finite_values), np.flatnonzero(~non_finite_values), arr[1:-1]['y'][~non_finite_values])
+    
     # decide which points are connected by lines
     if eq(connect, 'all'):
         arr[1:-1]['c'] = 1
-    elif eq(connect, 'pairs'):
-        arr[1:-1]['c'][::2] = 1
-        arr[1:-1]['c'][1::2] = 0
-    elif eq(connect, 'finite'):
-        arr[1:-1]['c'] = np.isfinite(x) & np.isfinite(y)
+        if non_finite_values[-1] and len(non_finite_values) > 1:
+            arr[1:-1]['c'][-2] = 0
     elif isinstance(connect, np.ndarray):
         arr[1:-1]['c'] = connect
     else:
-        raise Exception('connect argument must be "all", "pairs", "finite", or array')
+        if eq(connect, 'pairs'):
+            arr[1:-1]['c'][::2] = 1
+            arr[1:-1]['c'][1::2] = 0
+        elif eq(connect, 'finite'):
+            arr[1:-1]['c'] = 1
+        else:
+            raise Exception('connect argument must be "all", "pairs", "finite", or array')
+        
+        arr[1:-1]['c'][non_finite_values] = 0
+        arr[1:-1]['c'][np.roll(non_finite_values,-1)] = 0
```

Comment:
I think there's a bug here--roll(-1) moves the first element to the end of the array ?

Reply:
The `roll` was used because the last entry of `arr[1:-1]['c']` doesn't matter. Code to test:
```python3
import numpy as np
import pyqtgraph as pg

app = pg.mkQApp()

data = np.random.normal(size=10)
data2 = np.copy(data)
data2[0] = np.nan

pg.plot(data, title="no NaN")
pg.plot(data2, title="one NaN")

app.exec_()
```